backend/src/core/services/campaign.service.ts

```diff
@@ -188,9 +213,11 @@ const listCampaigns = ({
 
   if (status) {
     let operation: any = {}
```

Comment:
might be out of scope for this PR, but do you think you can refactor `listCampaigns` by building on [the PR here](https://github.com/opengovsg/postmangovsg/pull/1860) — I think the PR there is mostly done, can just copy and paste and add your changes

the current `listCampaigns` is really quite bad code 😅

Reply:
suggest leaving out of this PR first